src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentInvoker.cs

```diff
@@ -129,9 +141,9 @@ private IViewComponentResult InvokeSyncCore(ViewComponentContext context)
                 object result;
                 try
                 {
-                    result = method.Invoke(component, arguments);
+                    result = methodExecutor.Execute(component, arguments);
                 }
-                catch (TargetInvocationException ex)
+                catch (Exception ex)
                 {
                     _viewComponentFactory.ReleaseViewComponent(context, component);
```

Comment:
Same comment as the other code. This doesn't need a try/catch. The call to `_viewComponentFactory.ReleaseViewComponent(context, component);` should be in a `finally` though! Looks like this was a bug.